connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -448,14 +448,14 @@ public boolean isDeleted() {
             return deleted;
         }
 
-        public boolean canWrite(T status) {
-            return value != null &&
```

Comment:
The `value != null` condition was contrary to the motivation stated in this class of allowing for a 'safe' write to happen if there is no previous value.